File: frontend/biblioteca-app/src/app/models/livro.ts
export interface Livro {
  id: number;
  titulo: string;
  autorId: number;
  generoId: number;
  publicacao?: string | null;
  autorNome?: string;
  generoNome?: string;
  createdAt?: string;
  updatedAt?: string;
}

export type LivroUpsert = {
  titulo: string;
  autorId: number;
  generoId: number;
  publicacao?: string | null;
};

export type CreateLivro = LivroUpsert;

File: backend/src/Biblioteca.Domain/Entities/Livro.cs
using Biblioteca.Domain.Common;

namespace Biblioteca.Domain.Entities;

public class Livro : BaseEntity
{
    public string Titulo { get; set; } = default!;
    public int AutorId { get; set; }
    public Autor Autor { get; set; } = default!;
    public int GeneroId { get; set; }
    public Genero Genero { get; set; } = default!;
    public DateTime? Publicacao { get; set; }

}
